### DISCUSSION TITLE
Custom build folder

### DISCUSSION THREAD

**Discussion by preetham05** [microsoft/vscode-cmake-tools#2349] [General]
I am using cmake extension in vscode for building my embedded applications. My application needs to be build for multiple embedded targets. Is it possible to place all my build outputs in a separate folder specific to each board I build.. For example when I specify my board name using -DBOARD during compile time, could cmake extension create a folder under build for the board and place all the build files there?

**Comment by bobbrow** [comment]
This should be possible regardless of whether you use [CMake Presets](https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/cmake-presets.md) or [Kits + Variants](https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/variants.md).

With presets, you would set the `"buildDirectory"` property to something like: `"${sourceDir}/build/${presetName}"`
With Kits + Variants, you would set the `"cmake.buildDirectory"` setting to something like: `"${workspaceFolder}/build/${variant:Board}"` if you created a variants called "Board" that provides the options to switch between.
